src/EditorFeatures/Test/Outlining/OutliningTaggerTests.cs

```diff
@@ -133,16 +135,16 @@ private static List<IOutliningRegionTag> GetTagsFromWorkspace(TestWorkspace work
             var editorService = workspace.GetService<IEditorOptionsFactoryService>();
             var projectionService = workspace.GetService<IProjectionBufferFactoryService>();
 
-            var provider = new OutliningTaggerProvider.TagProducer(
-                textService, editorService, projectionService);
+            var provider = new OutliningTaggerProvider(
+                workspace.ExportProvider.GetExportedValue<IForegroundNotificationService>(),
+                textService, editorService, projectionService,
+                AggregateAsynchronousOperationListener.EmptyListeners);
 
-            Document document = workspace.CurrentSolution.GetDocument(hostdoc.Id);
+            var document = workspace.CurrentSolution.GetDocument(hostdoc.Id);
+            var context = new TaggerContext<IOutliningRegionTag>(document, view.TextSnapshot);
+            provider.ProduceTagsAsync(context).Wait();
```

Comment:
Can we have a single unit test helper in the base that does this for everyone to use? This fixes my other concern in that it requires everybody to make them public and comment "for testing only"